crates/uv/src/commands/tool/run.rs

```diff
@@ -420,15 +448,50 @@ async fn get_or_create_environment(
     cache: &Cache,
     printer: Printer,
     preview: PreviewMode,
-) -> Result<(Requirement, PythonEnvironment), ProjectError> {
+) -> Result<(ToolRequirement, PythonEnvironment), ProjectError> {
     let client_builder = BaseClientBuilder::new()
         .connectivity(connectivity)
         .native_tls(native_tls)
         .allow_insecure_host(allow_insecure_host.to_vec());
 
     let reporter = PythonDownloadReporter::single(printer);
 
-    let python_request = python.map(PythonRequest::parse);
+    // Check if the target is `python`
+    let python_request = if target.is_python() {
+        let target_request = match target {
+            Target::Unspecified(_) => None,
+            Target::Version(_, version) | Target::FromVersion(_, _, version) => {
+                Some(PythonRequest::Version(
+                    VersionRequest::from_str(&version.to_string()).map_err(anyhow::Error::from)?,
+                ))
+            }
+            // TODO(zanieb): Add `PythonRequest::Latest`
+            Target::Latest(_) | Target::FromLatest(_, _) => {
+                return Err(anyhow::anyhow!(
+                    "Requesting the 'latest' Python version is not yet supported"
+                )
+                .into())
+            }
+            // From the definition of `is_python`, this can only be a bare `python`
+            Target::From(_, from) => {
+                debug_assert_eq!(*from, "python");
+                None
+            }
+        };
+
+        if let Some(target_request) = &target_request {
+            if let Some(python) = python {
+                return Err(anyhow::anyhow!(
+                    "Received multiple Python version requests: `{python}` and `{target_request}`"
+                )
+                .into());
```

Comment:
This is like, `uvx --python 3.7 python@3.7`?

Reply:
Yeah. We could check for compatibility or merge them or whatever because `uvx --python pypy python@3.7` is "fine" but I don't do anything complicated here.